### PR TITLE
CSR-2: Reactive CRUD repository

### LINKED ISSUE #1: CRS-2: MongoDB Rest CRUD API
As a developer, I want to create a MongoDB Rest CRUD API to connect basic functionality to a project.
**Additional context**
==
How to use API:
==
Method: POST 
Link: /api/courses
Description: add course to DB.
Example JSON:
{
    "title": "ELK",
    "description": "ELK Tutorial"
}
==
Method: GET
Link: /api/courses
Description: Get all courses from DB.
==
Method: GET
Link: /api/courses/id
Description: Get course by ID.
==
Method: PUT
Link: /api/courses/id
Description: Update course by ID.
==
Method: DELETE
Link: /api/courses/id
Description: Delete course by ID.
==
Method: DELETE
Link: /api/courses/
Description: Delete all courses.
==
Method: GET
Link: /api/courses?title=[keyword]
Description: Find course by keyword in title.
==
Method: GET
Link: /api/courses/published
Description: get list of published courses.
==

### PR DESCRIPTION
**Pull request by Tim0f3y**

### CODE REVIEW

File: src/main/java/ru/jma/jmacourseservice/JmaCoursesServiceApplication.java [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
 import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Primary;
 
 @SpringBootApplication
 @EnableDiscoveryClient
 public class JmaCoursesServiceApplication {
 
+    @Bean
+    @Primary
```
Comment (FanilBiktimirov) [inline src/main/java/ru/jma/jmacourseservice/JmaCoursesServiceApplication.java]: Need to add this annotation?

File: src/main/java/ru/jma/jmacourseservice/JmaCoursesServiceApplication.java [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+    public ObjectMapper objectMapper() {
```
Comment (FanilBiktimirov) [inline src/main/java/ru/jma/jmacourseservice/JmaCoursesServiceApplication.java]: Create configuration class

File: src/main/java/ru/jma/jmacourseservice/controller/CourseController.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/courses")
+public class CourseController {
+
+    private final CourseService courseService;
+    private final ObjectMapper objectMapper;
+
+    @GetMapping
+    public Flux<CourseDto> getAllCourses(@RequestParam(required = false) String name) {
```
Comment (FanilBiktimirov) [inline src/main/java/ru/jma/jmacourseservice/controller/CourseController.java]: What is it name?

File: src/main/java/ru/jma/jmacourseservice/controller/CourseController.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import ru.jma.jmacourseservice.service.CourseService;
+
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/courses")
+public class CourseController {
+
+    private final CourseService courseService;
+    private final ObjectMapper objectMapper;
```
Comment (FanilBiktimirov) [inline src/main/java/ru/jma/jmacourseservice/controller/CourseController.java]: Try to use Mapstruct

File: src/main/java/ru/jma/jmacourseservice/service/CourseService.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import reactor.core.publisher.Mono;
+import ru.jma.jmacourseservice.model.Course;
+import ru.jma.jmacourseservice.repository.CourseRepository;
+
+@RequiredArgsConstructor
+@Service
+public class CourseService {
+    private final CourseRepository courseRepository;
+
+
```
Comment (FanilBiktimirov) [inline src/main/java/ru/jma/jmacourseservice/service/CourseService.java]: Formatting

**Review by FanilBiktimirov — CHANGES_REQUESTED** [review]

**Review by ilya2236 — APPROVED** [review]

**Review by FanilBiktimirov — APPROVED** [review]

**Pull request merged**